src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs

```diff
@@ -125,8 +124,19 @@ private long GetRawElapsedTicks()
                 // the Stopwatch is started last time.
                 long currentTimeStamp = GetTimestamp();
                 long elapsedUntilNow = currentTimeStamp - _startTimeStamp;
-                timeElapsed += elapsedUntilNow;
+
+                if (elapsedUntilNow > 0)
+                {
+                    // When measuring small time periods the Stopwatch.Elapsed*
+                    // properties can return negative values.  This is due to
+                    // bugs in the basic input/output system (BIOS) or the hardware
+                    // abstraction layer (HAL) on machines with variable-speed CPUs
+                    // (e.g. Intel SpeedStep).
+
+                    timeElapsed += elapsedUntilNow;
+                }
```

Comment:
We utilize the monotonic timers available from the underlying OS (`QueryPerformanceCounter` on Windows, `CLOCK_UPTIME_RAW` or `CLOCK_MONOTONIC` on Unix) which have strict contracts around not going backwards,

We can have up to a 1ns resolution, in which case overflow would take ~292 years in the worst possible scenario.

Given the support lifecycle of .NET and pieces of hardware, I'd say you have many other problems if you've not restarted your machine in 292 years ;)

Reply:
The comment here (copied from the other place above) relates to much older fallback behavior that existed for very old systems before `QPC` existed (back when we supported cases like Win98 or early Win2k machines).